### ISSUE TITLE
Unable to wrap the store in Edge

### ISSUE THREAD

**Issue by ChrisLMerrill** [tshaddix/webext-redux#199]
Wrapping the store in the background script in Edge fails with this message:

`SCRIPT5007: SCRIPT5007: Unable to get property 'onMessage' of undefined or null reference`

To reproduce:
- clone: https://github.com/ChrisLMerrill/CrossBrowserWebextensionReactReduxExample.git
- run: webextension-toolbox build edge
- install the extension from dist/edge
- look in the console for the background script

I'm not sure if this next bit is a side effect or another related problem: Connecting to the store from my pop-up script results in this message:

` SCRIPT5007: SCRIPT5007: Unable to get property 'connect' of undefined or null reference`

You can see this in the same example, by opening the pop-up via the extension button and checking the console.

TIA!

**Comment by ChrisLMerrill** [comment]
Looking at util.js, getBrowserAPI does this:

`var api = global.chrome || global.browser || browser;`

Looking at _global.chrome_, at that point it exists in Edge, it doesn't have much of anything attached to it. However, _browser_ does. So I changed it to:

`var api = browser || global.chrome || global.browser;`

and it works in Edge. However, this introduces a breakage in Chrome.   :(

So I have to ask why the code needs to do this at all? Why not just reference _browser_ and rely on the user to include a polyfill for browser compatibility? Or am I missing something else important?  (I am a relatively newbie with extension dev).

Chris

**Comment by tshaddix** [comment]
Hey @ChrisLMerrill  - sorry for the late response. I've been traveling for work!

So the main reason I structured it this way originally was for safety. The browser api calls we are making are still all callback based, which is a chrome specific implementation. Firefox and Edge support it, but they don't make any commitments to maintaining a callback-based API in the future. For that reason, I opted to use `chrome` as the initial default and `browser` as a fallback until all of the internal mechanisms can be switched to the promise API.
